components/Profile/Rank.tsx

```diff
@@ -1,30 +1,18 @@
-import getProfileRank from '@/lib/sonata/getProfileRank';
-import { useProfileProvider } from '@/providers/ProfileProvider';
-import { useEffect, useState } from 'react';
+import useRank from '@/hooks/useRank';
 
 const Rank = () => {
-  const [rank, setRank] = useState(0);
-  const { profile } = useProfileProvider();
+  const rank = useRank();
 
-  useEffect(() => {
-    const init = async () => {
-      if (!profile?.verifications) return;
-      const leaderBoardRank: any = await getProfileRank(profile?.verifications);
-      setRank(leaderBoardRank);
-    };
-    if (!profile) return;
-
-    init();
-  }, [profile]);
+  if (rank === null) {
+    return null;
```

Comment:
please do not return `empty` components.

instead, handle this login in the parent to `conditionally render` the `<Rank />` component

Reply:
This has been fixed.